src/monexamine.cpp

```diff
@@ -215,6 +256,62 @@ int monexamine::pet_armor_pos( monster &z )
     return pos;
 }
 
+void monexamine::remove_battery( monster &z )
+{
+    g->m.add_item_or_charges( g->u.pos(), *z.battery_item );
+    z.battery_item = cata::nullopt;
+}
+
+void monexamine::insert_battery( monster &z )
+{
+    std::vector<item *> bat_inv = g->u.items_with( []( const item & itm ) {
+        return itm.has_flag( "MECH_BAT" );
+    } );
+    if( bat_inv.empty() ) {
+        return;
+    }
+    int i = 0;
+    uilist selection_menu;
+    selection_menu.text = string_format( _( "Select an battery to insert into your %s." ),
+                                         z.get_name() );
+    selection_menu.addentry( i++, true, MENU_AUTOASSIGN, _( "Cancel" ) );
+    for( auto iter : bat_inv ) {
+        selection_menu.addentry( i++, true, MENU_AUTOASSIGN, _( "Use %s" ), iter->tname() );
+    }
+    selection_menu.selected = 1;
+    selection_menu.query();
+    auto index = selection_menu.ret;
+    if( index == 0 || index == UILIST_CANCEL || index < 0 ||
+        index > static_cast<int>( bat_inv.size() ) ) {
+        return;
+    }
+    auto bat_item = bat_inv[index - 1];
+    int item_pos = g->u.get_item_position( bat_item );
+    if( item_pos != INT_MIN ) {
+        z.battery_item = *bat_item;
```

Comment:
What happens with the battery that was previously there?

Reply:
Added an early return for insert_battery() if the mech already has one.

I want seperate removal and insertion functions, because refurbishing/recharging a removed battery may be possible/useful in future, and you may want to remove the battery without necessarily inserting a new one to replace it.